src/services/observables/balances.ts

```diff
@@ -20,29 +23,26 @@ export const fetchTokenBalances = async (
     tokens,
     (token) => token.address === ethToken
   );
-  const shapes = tokensNoETH.map((x) => balanceShape(x.address, user));
+
+  const calls = tokensNoETH.map((x) => buildTokenBalanceCall(x.address, user));
 
   try {
     const [tokenBalances, ethBalance]: [
-      TokenBalance[][] | undefined,
+      Result[] | undefined,
       string | undefined
     ] = await Promise.all([
-      multi({
-        groupsOfShapes: [shapes],
-        currentNetwork,
-      }),
+      multicall(currentNetwork, calls),
       eth && fetchETH(user),
     ]);
-
-    if (tokenBalances && tokenBalances.length > 0) {
-      const balances = tokenBalances[0].map((token) => {
-        const inedx = tokensNoETH.findIndex((t) => t.address === token.address);
+    if (tokenBalances) {
+      const balances = tokenBalances.map((bn, index) => {
+        const balance = (bn[0] as BigNumber).toString();
```

Comment:
why not properly defining `balance` here instead of later doing a double trinary operator?